management_api_app/api/routes/workspace_service_templates.py

```diff
@@ -29,3 +34,21 @@ async def create_workspace_service_template(
         return template
     except EntityVersionExist:
         raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=strings.WORKSPACE_TEMPLATE_VERSION_EXISTS)
+
+
+@router.post("/workspace-service-templates/{template_name}/user-resource-templates",
+             status_code=status.HTTP_201_CREATED,
+             response_model=UserResourceTemplateInResponse, name=strings.API_CREATE_USER_RESOURCE_TEMPLATES)
+async def register_user_resource_template(
+        user_resource_template_create: UserResourceTemplateInCreate,
+        user_resource_template_repo: UserResourceTemplateRepository = Depends(get_repository(UserResourceTemplateRepository)),
+        workspace_service_template: ResourceTemplate = Depends(get_workspace_service_template_by_name_from_path)
+) -> UserResourceTemplateInResponse:
+    try:
+        template_created = create_user_resource_template(user_resource_template_create,
+                                                         user_resource_template_repo,
+                                                         workspace_service_template.name)
+        enriched_template = enrich_workspace_service_schema_defs(template_created)
```

Comment:
should we not save the enriched version in the database? - or do we always have to enrich on retrieval?

Reply:
We always enrich on retrieval

Reply:
we don't have to but we do. We can put an lru cache on it if speed is of essence. Don't think is should go to the DB because there is no enriched model in the domain. We can create one ofcourse but it does not have a physical meaning as the enrichment if purely for implementation needs.